docs/DeveloperGuide.md

```diff
@@ -4,30 +4,274 @@
+
+### Event component
+
+The `Event` component and its component classes are shown in the below **Class Diagram**:
+
+<img src = "images/EventClassDiagram.png">
+
+The `Event` component does the following:
+* Handle the addition, removal and marking of events stored in `EventList`.
+* Add, remove and mark participants for a specific `Event` in `EventList`.
+
+## Implementation
+
+This section describes some noteworthy details on how certain features are implemented.
+
+### List feature[TBD]
+
+The `list` feature allows users to view all scheduled events in the system. 
+It is implemented in the `ListCommand` class, which extends the base `Command` class and formats the output to display all events. 
+Internally, the `list` operation does not modify any data but simply retrieves and displays the information from `EventList`.
+
+The `ListCommand` class performs the following key operations:
+
+* ListCommand#execute() — Generates a formatted message displaying all events in the list.
+
+These operations are accessible through the `Command` and can be invoked when the list command is entered by the user.
+
+#### Feature Implementation
+
+Given below is an example usage scenario and the behavior of the list feature at each step:
+
+1. User Command Input:
+The user enters the command `list` to view all scheduled events.
+
+2. Command Recognition:
+The `COMMAND_WORD` is set to "list", enabling the system to recognize the command input and invoke `ListCommand`.
+
+3. Execution of ListCommand#execute():
+The `execute()` method retrieves each event from `eventList` and appends it to a formatted output message.
+It uses the `String.format` method with `LIST_MESSAGE` to include the total number of events in the message header.
+Events are appended to `outputMessage` with numbered formatting for readability.
+
+4. Output Generation:
+The method stores the generated `outputMessage` in `this.message`, ready for display.
+
+### Add feature
+
+The `add` feature allow users to add events/participants based on relevant flags.
+It is implemented in the `AddCommand` class which extends the base `Command` class and parse through the command to retrieve information based off flags.
+
+The above operation is implemented as `AddCommand#execute()`. This overrides the `Command#execute()` operation in `Command`,
+and is invoked when the latter operation is called.
+
+The `AddCommand` handles two primary functions:
+
+1. **Add an Event:** When provided with event details, including name, time, venue and priority, it creates and stores a new event in the event list.
+2. **Add a Participant:** When provided with participant information, including name, contact number, and email, it attempts to add the participant to an existing event.
+
+#### Feature Implementation
+
+Given below is an example usage scenario for the `add` mechanism, and how it behaves at each step.
+
+1. The user enters the command `add` followed by `-e` or `-p` to indicate adding an event or participant.
```

Comment:
Does the developer need to know exactly what flags are being used to execute this command? Can it be generalized/explained in pseudo terms?